msticpy/data/azure_data.py

```diff
@@ -226,22 +226,15 @@ def get_resources(  # noqa: MC0001
 
         self._check_client("resource_client", sub_id)
 
-        # pylint: disable=unnecessary-comprehension
-        try:
-            [r for r in self.resource_client.resources.list()]  # type: ignore
-        except AttributeError:
-            self._legacy_auth("resource_client", sub_id)
-
+        resources = []  # type: List
         if rgroup is None:
-            # pylint: disable=line-too-long
-            resources = [resource for resource in self.resource_client.resources.list()]  # type: ignore
```

Comment:
What was wrong with the list comprehension? - they are much faster than for loops. Sourcery prefers them

Reply:
So that's what I thought but pylint throws this whenever I used list comprehension here:
![image](https://user-images.githubusercontent.com/8940401/101139425-6f02eb80-3609-11eb-80bd-3fdbf26289cf.png)

Reply:
In this particular line it is unnecessary because it looks like self.resource_client.resources.list() is already a list (so the list comp is not doing anything)